core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +121,26 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  /**
+   * Traverses the history of the table's current snapshot and finds the snapshot with the given snapshot id as its
+   * parent.
+   * @return null if the passed in snapshot is not present in the table, else the snapshot for which the given snapshot
+   * is the parent
+   * @throws IllegalArgumentException when the given SnapshotId is not found in the table
+   * @throws IllegalStateException when the given snapshot ID is not an ancestor of the current table state
+   */
+  public static Snapshot snapshotAfter(Table table, long snapshotId) {
+    Snapshot previousSnapshot = table.snapshot(snapshotId);
+    Preconditions.checkArgument(previousSnapshot != null,
+        "Invalid snapshotId: %s, snapshot not found in the table.", snapshotId);
```

Comment:
The ID isn't invalid, the problem is that the snapshot is not present. I think it should be `"Cannot find parent snapshot: %s"`